microcosm_pubsub/tests/test_tracing.py

```diff
@@ -0,0 +1,128 @@
+import json
+from unittest.mock import ANY, patch
+
+from hamcrest import assert_that, equal_to
+from microcosm.api import create_object_graph
+
+from microcosm_pubsub import tracing
+from microcosm_pubsub.message import SQSMessage
+from microcosm_pubsub.result import MessageHandlingResultType
+from microcosm_pubsub.tests.fixtures import DerivedSchema, ExampleDaemon
+
+
+def test_trace_message_publish():
+    def loader(metadata):
+        return dict(
+            sns_topic_arns=dict(
+                default="topic-name",
+            )
+        )
+
+    graph = create_object_graph("example", testing=True, loader=loader)
+    graph.sns_producer.sns_client.publish.return_value = dict(MessageId="message-id-1234")
+
+    with patch.object(tracing, "oneagent") as oneagent, \
+            patch.object(tracing, "ChannelType"), \
+            patch.object(tracing, "Channel") as channel, \
+            patch.object(tracing, "MessagingDestinationType") as messagingdestinationtype:
+        sdk = oneagent.get_sdk.return_value
+        tracer_manager = sdk.trace_outgoing_message.return_value
+        tracer_obj = tracer_manager.__enter__.return_value
+        tracer_obj.outgoing_dynatrace_string_tag = "tag-1234"
+
+        message_id = graph.sns_producer.produce(
+            DerivedSchema.MEDIA_TYPE,
+            data="data",
+            opaque_data={"X-Request-Id": "req-id-1234"}
+        )
+        assert_that(message_id, equal_to("message-id-1234"))
+
+        graph.sns_producer.sns_client.publish.assert_called_once_with(
+            Message=ANY,
+            MessageAttributes=dict(
+                media_type=dict(
+                    DataType='String',
+                    StringValue='application/vnd.microcosm.derived'
+                ),
+            ),
+            TopicArn='topic-name'
+        )
+        message = json.loads(graph.sns_producer.sns_client.publish.call_args[1]['Message'])
+        assert_that(message["opaqueData"]["X-Request-Id"], equal_to("req-id-1234"))
```

Comment:
would the usual `assert` instead `assert_that` work in this file? just wondering if you did this intentional.

Reply:
gotcha! I did feel this was intentional 👍🏾  thank you for explaining!